src/main/java/com/amazon/opendistroforelasticsearch/knn/plugin/script/KNNScoringUtil.java

```diff
@@ -34,18 +37,55 @@
     public static float l2Squared(float[] queryVector, float[] inputVector) {
         float squaredDistance = 0;
         for (int i = 0; i < inputVector.length; i++) {
-            float diff = queryVector[i]-inputVector[i];
+            float diff = queryVector[i] - inputVector[i];
             squaredDistance += diff * diff;
         }
         return squaredDistance;
     }
 
+    private static float[] toFloat(List<Number> inputVector) {
+        Objects.requireNonNull(inputVector);
+        float[] value = new float[inputVector.size()];
+        int index = 0;
+        for (final Number val : inputVector) {
+            value[index++] = val.floatValue();
+        }
+        return value;
+    }
+
+    /**
+     * Whitelisted l2Squared method for users to calculate L2 squared distance between query vector
+     * and document vectors
+     * Example
+     *  "script": {
+     *         "source": "1/(1 + l2Squared(params.query_vector, doc[params.field]))",
+     *         "params": {
+     *           "query_vector": [1, 2, 3.4],
+     *           "field": "my_dense_vector"
+     *         }
+     *       }
+     *
+     * @param queryVector query vector
+     * @param docValues   script doc values
+     * @return L2 score
+     */
+    public static float l2Squared(List<Number> queryVector, KNNVectorScriptDocValues docValues) {
+        float[] knnDocVector;
+        try {
+            knnDocVector = docValues.getValue();
+        } catch (Exception e) {
```

Comment:
Should we catch the specific exception here?

Reply:
getValue can throw RuntimeException & IOException, i see three option here
1) Let docValues throw exception as it is and catch all individual exception here and return Float.Min like now,
2) Update getValue to throw only RuntimeException, and catch only RuntimeException here
3) Just update catch to catch ( RuntimeException | IOException e )

i don't see any difference in all three since at the end outcome is same, but if you strong opinion on one vs other, i can change it . if you see any other option i can make change it as well.

Reply:
Makes sense, Option 1 us good with me